go/tasks/plugins/k8s/sagemaker/custom_training.go

```diff
@@ -0,0 +1,184 @@
+package sagemaker
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	trainingjobController "github.com/aws/amazon-sagemaker-operator-for-k8s/controllers/trainingjob"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	flyteIdlCore "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	taskError "github.com/lyft/flyteplugins/go/tasks/errors"
+	pluginsCore "github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/ioutils"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/k8s"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/utils"
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/pkg/errors"
+
+	commonv1 "github.com/aws/amazon-sagemaker-operator-for-k8s/api/v1/common"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/k8s/sagemaker/config"
+
+	trainingjobv1 "github.com/aws/amazon-sagemaker-operator-for-k8s/api/v1/trainingjob"
+	flyteSageMakerIdl "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins/sagemaker"
+)
+
+func (m awsSagemakerPlugin) buildResourceForCustomTrainingJob(
+	ctx context.Context, taskCtx pluginsCore.TaskExecutionContext) (k8s.Resource, error) {
+
+	logger.Infof(ctx, "Building a training job resource for task [%v]", taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName())
+	taskTemplate, err := getTaskTemplate(ctx, taskCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal the custom field of the task template back into the Hyperparameter Tuning Job struct generated in flyteidl
+	sagemakerTrainingJob := flyteSageMakerIdl.TrainingJob{}
+	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &sagemakerTrainingJob)
```

Comment:
we should just check that custom is nil, or maybe add that as a common check in UnmarshalStruct

Reply:
It is nil-checked in UnmarshalStruct already